src/prefect/engine/task_runner.py

```diff
@@ -982,6 +977,7 @@ def check_for_retry(self, state: State, inputs: Dict[str, Result]) -> State:
             run_count = prefect.context.get("task_run_count", 1)
             if prefect.context.get("task_loop_count") is not None:
                 loop_context = {
+                    # todo: use task's converted result object?
                     "_loop_count": Result(
```

Comment:
These need to be freshly instantiated JSONResults, not dependent on the user's Result subclass that they configured